Develop an SQL schema for SOPs

```diff
@@ -1,29 +1,4351 @@
+--
+
+ALTER TABLE ONLY public."INVESTIGATION"
+    ADD CONSTRAINT "INVESTIGATION_pkey" PRIMARY KEY ("Investigation_ID");
+
+
+--
+-- TOC entry 2782 (class 2606 OID 17688)
+-- Name: ONTOLOGY_SOURCE_REFERENCE ONTOLOGY_SOURCE_REFERENCE_Ontology_REF_Investigation_Identi_key; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."ONTOLOGY_SOURCE_REFERENCE"
+    ADD CONSTRAINT "ONTOLOGY_SOURCE_REFERENCE_Ontology_REF_Investigation_Identi_key" UNIQUE ("Ontology_REF_Investigation_Identifier");
+
+
+--
+-- TOC entry 2784 (class 2606 OID 17514)
+-- Name: ONTOLOGY_SOURCE_REFERENCE ONTOLOGY_SOURCE_REFERENCE_pkey; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."ONTOLOGY_SOURCE_REFERENCE"
+    ADD CONSTRAINT "ONTOLOGY_SOURCE_REFERENCE_pkey" PRIMARY KEY ("Ontology_Source_REF_ID");
+
+
+--
+-- TOC entry 2786 (class 2606 OID 17525)
+-- Name: PROTOCOLS PROTOCOLS_pkey; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."PROTOCOLS"
+    ADD CONSTRAINT "PROTOCOLS_pkey" PRIMARY KEY ("Protocol_ID");
+
+
+--
+-- TOC entry 2788 (class 2606 OID 17536)
+-- Name: PUBLICATIONS PUBLICATIONS_pkey; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."PUBLICATIONS"
+    ADD CONSTRAINT "PUBLICATIONS_pkey" PRIMARY KEY ("Publication_ID");
+
+
+--
+-- TOC entry 2794 (class 2606 OID 17559)
+-- Name: STUDY_DESIGN_DESCRIPTORS STUDY_DESIGN_DESCRIPTORS_pkey; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."STUDY_DESIGN_DESCRIPTORS"
+    ADD CONSTRAINT "STUDY_DESIGN_DESCRIPTORS_pkey" PRIMARY KEY ("Design_Descriptors_ID");
+
+
+--
+-- TOC entry 2796 (class 2606 OID 17570)
+-- Name: STUDY_FACTORS STUDY_FACTORS_pkey; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."STUDY_FACTORS"
+    ADD CONSTRAINT "STUDY_FACTORS_pkey" PRIMARY KEY ("Study_Factors_ID");
+
+
+--
+-- TOC entry 2798 (class 2606 OID 17578)
+-- Name: STUDY_SAMPLES STUDY_SAMPLES_pkey; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."STUDY_SAMPLES"
+    ADD CONSTRAINT "STUDY_SAMPLES_pkey" PRIMARY KEY ("Sample_Name");
+
+
+--
+-- TOC entry 2790 (class 2606 OID 17681)
+-- Name: STUDY STUDY_Study_Identifier_key; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."STUDY"
+    ADD CONSTRAINT "STUDY_Study_Identifier_key" UNIQUE ("Study_Identifier");
+
+
+--
+-- TOC entry 2792 (class 2606 OID 17548)
+-- Name: STUDY STUDY_pkey; Type: CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."STUDY"
+    ADD CONSTRAINT "STUDY_pkey" PRIMARY KEY ("Study_ID");
+
+
+--
+-- TOC entry 2805 (class 2606 OID 17663)
+-- Name: ASSAY ASSAY_Assay_ID_fkey; Type: FK CONSTRAINT; Schema: public; Owner: postgres
+--
+
+ALTER TABLE ONLY public."ASSAY"
+    ADD CONSTRAINT "ASSAY_Assay_ID_fkey" FOREIGN KEY ("Assay_ID") REFERENCES public."A_TECHNOLOGY_MICROARRAY"("Assay_ID") NOT VALID;
```

Comment:
Are these foreign keys in the right direction?  I am thinking the Assay_ID in the `A_TECHNOLOGY_...` tables should reference the `ASSAY` table just like what you have below for the samples.

Reply:
I corrected it